lib/PlacesProvider.js

```diff
@@ -668,6 +669,88 @@ Links.prototype = {
   }),
 
   /**
+   * Gets the remote tabs links.
+   *
+   * @param {Integer} options
+   *
+   *
+   * @returns {Promise} Returns a promise with the array of links as payload.
+   */
+  getRemoteTabsLinks:  Task.async(function*(options = {}) {
+    let {limit} = options;
+    if (!limit || limit.options > LINKS_QUERY_LIMIT) {
+      limit = LINKS_QUERY_LIMIT;
+    }
+
+    // We first get the URLs open on other devices from SyncedTabs.
+    let clients = yield SyncedTabs.getTabClients();
+
+    if (clients.length === 0) {
+      // No reason to waste cycles doing anything else.
+      return;
+    }
+
+    // get the URLs with the data we need from all the clients
+    let urls = {};
+    for (let client of clients) {
+      for (let tab of client.tabs) {
+        urls[tab.url] = {
+          deviceName: client.name,
+          deviceIsMobile: client.isMobile,
+          lastUsed: tab.lastUsed
+        };
+      }
+    }
+
+    // Now we add the data required from the places DB.
+    // setup binding parameters
+    let params = {
+      limit: limit
+    };
+
+    let blockedURLs = this.blockedURLs.items().map(item => `"${item}"`);
```

Comment:
It can be an empty list. The query still works although maybe it isn't optimal to still have the `moz_places.url NOT IN (${blockedURLs})` in the SQL query. I just followed what all the other queries in this file are doing, but it might be worth following up on the perf of that.